src/main/scala/cognite/spark/v1/FlexibleDataModelCorePropertySyncRelation.scala

```diff
@@ -0,0 +1,91 @@
+package cognite.spark.v1
+
+import cats.effect.{Async, IO}
+import cognite.spark.v1.FlexibleDataModelBaseRelation.ProjectedFlexibleDataModelInstance
+import cognite.spark.v1.FlexibleDataModelRelationFactory.{
+  ViewCorePropertyConfig,
+  ViewSyncCorePropertyConfig
+}
+import com.cognite.sdk.scala.common.ItemsWithCursor
+import com.cognite.sdk.scala.v1.GenericClient
+import com.cognite.sdk.scala.v1.fdm.common.filters.FilterDefinition.HasData
+import com.cognite.sdk.scala.v1.fdm.instances._
+import fs2.Stream
+import org.apache.spark.sql.SQLContext
+import org.apache.spark.sql.sources._
+
+/**
+  * Flexible Data Model Relation for syncing Nodes or Edges with properties
+  *
+  * @param config common relation configs
+  * @param corePropConfig view core property config
+  * @param sqlContext sql context
+  */
+private[spark] class FlexibleDataModelCorePropertySyncRelation(
+    config: RelationConfig,
+    corePropConfig: ViewSyncCorePropertyConfig)(sqlContext: SQLContext)
+    extends FlexibleDataModelCorePropertyRelation(
+      config,
+      ViewCorePropertyConfig(
+        corePropConfig.intendedUsage,
+        corePropConfig.viewReference,
+        corePropConfig.instanceSpace)
+    )(sqlContext) {
+
+  override def getStreams(filters: Array[Filter], selectedColumns: Array[String])(
+      client: GenericClient[IO]): Seq[Stream[IO, ProjectedFlexibleDataModelInstance]] = {
+    val selectedInstanceProps = if (selectedColumns.isEmpty) {
+      schema.fieldNames
+    } else {
+      selectedColumns
+    }
+
+    val syncRequests = compatibleInstanceTypes(intendedUsage).map { instanceType =>
+      val hasData: Option[HasData] = viewReference.map { viewRef =>
+        HasData(List(viewRef))
+      }
+      val tableExpression = instanceType match {
+        case InstanceType.Edge => TableExpression(edges = Some(EdgeTableExpression(filter = hasData)))
+        case InstanceType.Node => TableExpression(nodes = Some(NodesTableExpression(filter = hasData)))
+      }
+      val sourceReference: Seq[SourceSelector] = viewReference
+        .map(
+          r =>
+            SourceSelector(
+              source = r,
+              properties = selectedInstanceProps.toIndexedSeq.filter(p =>
+                p != "startNode" && p != "endNode" && p != "space" && p != "externalId" && p != "type")))
+        .toSeq
+
+      InstanceSyncRequest(
+        `with` = Map("sync" -> tableExpression),
+        cursors = None,
```

Comment:
Due to how jetfire and our sparksql bindings are looking like now, we will pass the sync cursor down with the set up of the datasource (as an option()). We will also limit ourselves to one cursor at a time, as there is no way I have found that you can specify multiple with clauses from Spark SQL. 

We need to pick up this cursor and propagate it here, I guess?

Reply:
Additional question is how to pass it out. And to double-check that cursor can be used multiple times in case job has to restart

Reply:
Passing it out will be independent of the spark data source (like done for cursors today), directly to the incremental version storage. 

Yes, the same cursors can be used multiple times.